### PR TITLE
chore: update this project to use the generator

### PR DESCRIPTION
**Pull request by brandonocasey**
> Note: This change is not a major version update, but **any** syntax updates in `lib` will require us to do a major.

~Based on #351 as that is a required, but easily separate piece to standardizing this project.~ has been merged

## Changes
* Segments are generated via rollup as we do in vhs, I compiled them back into binary files.
* Commented out web worker code in `test/metadata-stream.test.js` as it will take some work to get it working with rollup. See TODO below.

## TODO (in another pull request):
* ~Add a plugin to unit test the `WebWorker` code in `test/metadata-stream.test.js` that I commented out as it relies on browserify.~ Done via [rollup-plugin-worker-factory](https://github.com/brandonocasey/rollup-plugin-worker-factory)
* Update the code to linting standards in phases and reenable the `lint` script by removing the `:` from the front.
* Update the `pre-commit` hook to run `lint-staged` again by removing the `:` in-front of it.

### CODE REVIEW

File: scripts/rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  },
+  plugins(defaults) {
+    defaults.test.splice(0, 0, 'createTestData');
+
+    return defaults;
+  }
+};
+const mainBuilds = generate(Object.assign({input: 'lib/index.js', distName: 'mux'}, shared)).builds;
+const mp4Builds = generate({input: 'lib/mp4/index.js', distName: 'mux-mp4'}).builds;
+const flvBuilds = generate({input: 'lib/flv/index.js', distName: 'mux-flv'}).builds;
```
Comment (gkatsev) [inline scripts/rollup.config.js:21]: What's being used for the UMD name here? Looks like previously it was set to `muxjs`.

Comment (brandonocasey) [reply scripts/rollup.config.js]: updated to muxjs

File: test/utils/sei-nal-unit-generator.js [hunk 2]
```diff
@@ -81,9 +81,11 @@ var makeMdatFromCaptionPackets = function(packets) {
     mdat.push(sei.escapedRBSP.length + 1); // nal length
     mdat.push(0x06); // declare nal type as SEI
     // SEI message
-    sei.escapedRBSP.forEach(function(byte) {
+    for (var i = 0; i < sei.escapedRBSP.length; i++) {
```
Comment (brandonocasey) [inline test/utils/sei-nal-unit-generator.js:84]: small change as ie 11 does not support `forEach` on a `Uint8Array`

File: test/caption-stream.test.js [hunk 3]
```diff
@@ -2635,7 +2637,7 @@ QUnit.test('Cea608Stream will log errors, not throw an exception', function(asse
     'warnings were logged to the console'
   );
 
-  window.console.error = originalConsole;
+  console.error = originalConsole;
```
Comment (brandonocasey) [inline test/caption-stream.test.js:2640]: nodejs support

**Review by gkatsev — APPROVED** [review]
Tested with this linked into VHS, and it's working as expected.

**Pull request merged**